directories/src/redux/epics/node-epics.js

```diff
@@ -153,6 +161,47 @@ const requestGenesisHashEpic = (action$, store) => {
     });
 };
 
+const treasureHuntEpic = (action$, store) => {
+  return action$
+    .ofType(nodeActions.TREASURE_HUNT)
+    .mergeMap((action) => {
+      const { genesisHash, numberOfChunks } = action.payload;
+      const datamap = Datamap.generate(genesisHash, numberOfChunks);
+      const addresses = _.values(datamap);
+      const countSector = addresses / SECTOR_DIVIDER;
+      const randomSector = AppUtils.randomArray(1, countSector);
+      randomSector.map((sectorIndex) => {
+        let min = 0;
+        if((sectorIndex - 1) !== 0) {
+          min = (sectorIndex - 1) * SECTOR_DIVIDER;
+        }
+        const max = (sectorIndex * SECTOR_DIVIDER) - 1;
+        const randomSectorAddress = AppUtils.randomArray(min, max);
+        randomSectorAddress.map((addressIndex) => {
+          randomAddress = addresses.slice(addressIndex);
+          return Observable.fromPromise(
+            iota.findTransactions(randomAddress)
+          )
+            .mergeMap(({ hashes }) => {
+              return Observable.fromPromise(
+                iota.getTrytes(hashes)
+              )
+              .map(({ trytesArray }) => {
+                const transactionObject = iota.utils.TransactionObject(trytesArray);
+                const asciiTimestampTrytes = trytesArray.map(trytes => trytes.timestamp.charCodeAt(0))
+                  .reduce((current, previous) => previous + current);
+                return nodeActions.addGenesisHash({ genesisHash, numberOfChunks });
```

Comment:
What is the reasoning for calling addGenesisHash here?  And where does the actual treasure hunting occur?  i.e. building the sideChain, and attempting to unlock the "message" payload with the sha512 hashes?

Reply:
Ok, I know there have to be next actions but what is sidechain and what attemting message mean?